### ISSUE TITLE
JS function Files.isFileNameValid is out of date

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/gallery#241]
Gallery:
```js
	isFileNameValid: function (name) {
		"use strict";
		var trimmedName = name.trim();
		if (trimmedName === '.' || trimmedName === '..') {
			throw t('files', '"{name}" is an invalid file name.', {name: name});
		} else if (trimmedName.length === 0) {
			throw t('files', 'File name cannot be empty.');
		}
		return true;

	},
```
Files app:
```js
		isFileNameValid: function (name) {
			var trimmedName = name.trim();
			if (trimmedName === '.'	|| trimmedName === '..')
			{
				throw t('files', '"{name}" is an invalid file name.', {name: name});
			} else if (trimmedName.length === 0) {
				throw t('files', 'File name cannot be empty.');
			} else if (OC.fileIsBlacklisted(trimmedName)) {
				throw t('files', '"{name}" is not an allowed filetype', {name: name});
			}

			return true;
		},
```

`OC.fileIsBlacklisted` check is missing

**Comment by oparoz** [comment]
Thanks for the report :)

**Comment by MorrisJobke** [comment]
Fix is in #273
